samples/MetroDemo/MainWindowViewModel.cs

```diff
@@ -339,7 +340,32 @@ public ICommand ShowProgressDialogCommand
 
             await controller.CloseAsync();
         }
-        
+        private ICommand showCustomInputDialogCommand;
+
+        public ICommand ShowCustomInputDialogCommand
+        {
+            get
+            {
+                return this.showCustomInputDialogCommand ?? (this.showCustomInputDialogCommand = new SimpleCommand
+                {
+                    CanExecuteDelegate = x => true,
+                    ExecuteDelegate = x => RunCustomInputFromVm()
+                });
+            }
+        }
+
+        private async void RunCustomInputFromVm()
+        {
+            var customDialog = new PersonInputDialog() {Title = "Select Person"};
+
+            //var resultingPerson = _dialogCoordinator.ShowCustomInputAsync(this, customDialog).Result; //todo: this doesnt work
+            var resultingPerson = await _dialogCoordinator.ShowCustomInputAsync(this, customDialog);
+
+            if (resultingPerson != null) {
+                await _dialogCoordinator.ShowMessageAsync(this, "YOUR CHOICE",
+                    $"You selected {resultingPerson.Name}, who is aged {resultingPerson.Age}");
```

Comment:
this is a c# 6.0 feature which is currently not supported here in MahApps solution